internal/testutils/random.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package testutils
+
+import (
+	"hash/crc64"
+	"math/rand"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+)
+
+// The functions below contain an assortment of random ID generation functions, partially ported and improved from the
+// internal/legacy/helper/acctest package.
+
+var randomSources = map[string]*rand.Rand{} //nolint:gochecknoglobals //This variable stores the randomness sources for DeterministicRandomID and needs to be global.
+var randomLock = &sync.Mutex{}              //nolint:gochecknoglobals //This variable is required to lock the randomSources above.
+
+// CharacterSpace defines which characters to use for generating a random ID.
+type CharacterSpace string
+
+const (
+	CharacterSpaceAlphaNumeric      CharacterSpace = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
```

Comment:
"Space" in `CharacterSpace` makes me thing of `' '` . Maybe CharacterRange or similar?